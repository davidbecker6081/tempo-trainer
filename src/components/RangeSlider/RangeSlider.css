.rc-slider-rail,
.rc-slider-track,
.rc-slider-step {
  height: 20px;
}

.rc-slider-track {
  background-color: #4EE46F;
}

.rc-slider-rail {
  background-color: #252525;
}

.rc-slider-handle {
  height: 30px;
  width: 30px;
  border-color: #4EE46F;;
}

.current-range {
  display: flex;
  align-items: center;
  justify-content: space-around;
  margin-bottom: 1em;
  width: 100%;
  background-color: #252525;
  border-radius: 5px;
  box-shadow: 0px 0px 2px 2px #25252563;
  color: #FFFFFF;
  font-size: 1.5em;
}

.current-range-num {
  font-weight: 700;
}
