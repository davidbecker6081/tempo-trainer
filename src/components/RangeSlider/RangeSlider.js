import React from 'react';
import PropTypes from 'prop-types';
import { Range } from 'rc-slider';
import 'rc-slider/assets/index.css';
import './RangeSlider.css';

const RangeSlider = ({ originalRange, handleRangeChange, range }) => {
  const min = originalRange[0];
  const max = originalRange[1];
  const style = { width: '50vw', height: '5vh' };

  return (
    <div>
      <Range
        min={min}
        max={max}
        allowCross={false}
        onChange={e => handleRangeChange(e)}
        defaultValue={[min, max]}
        value={range}
        style={style}
        pushable={1}
      />
    </div>
  );
};


export default RangeSlider;

RangeSlider.propTypes = {
  handleRangeChange: PropTypes.func.isRequired,
  range: PropTypes.arrayOf(PropTypes.number).isRequired,
  originalRange: PropTypes.arrayOf(PropTypes.number).isRequired,
};
