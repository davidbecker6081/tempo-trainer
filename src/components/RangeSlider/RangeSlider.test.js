import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import RangeSlider from './RangeSlider';

describe('RangleSlider', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      originalRange: [],
      handleRangeChange: () => {},
      range: [],
    };
  });

  describe('originalRange is defined', () => {
    beforeEach(() => { mockProps.originalRange = [1, 5]; });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('range is defined', () => {
    beforeEach(() => { mockProps.range = [0, 5]; });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
