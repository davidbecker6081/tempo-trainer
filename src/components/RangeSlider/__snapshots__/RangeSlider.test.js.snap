// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeSlider originalRange is defined should render correclty 1`] = `undefined`;

exports[`RangeSlider range is defined should render correclty 1`] = `undefined`;
